dreamplace/PlaceDB.py

```diff
@@ -87,6 +87,11 @@ def __init__(self):
         self.bin_size_y = None
         self.num_bins_x = None
         self.num_bins_y = None
+        
+        # add bin center and non movable macros info for circuit training
+        self.bin_center_x = None
+        self.bin_center_y = None
```

Comment:
I'm trying to get rid of bin centers in the PlaceDB, as the density related OPs no longer need it. Do you really need to store them?

Reply:
Removed this part.